test/data/basic-job-with-template.yaml

```diff
@@ -0,0 +1,7 @@
+jobs:
+    main:
+        template: mytemplate@1.2.3
```

Comment:
Can we add some steps in the job to verify that the template steps override the job step?